File: android/app/src/main/java/com/golfiq/hud/analytics/AnalyticsConsentStore.kt
package com.golfiq.hud.analytics

import android.content.Context
import android.content.SharedPreferences

internal class AnalyticsConsentStore(context: Context) {
    companion object {
        private const val PREFS_NAME = "analytics_consent"
        private const val KEY_GRANTED = "consent_granted"
    }

    private val preferences: SharedPreferences =
        context.getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE)

    fun isGranted(): Boolean = preferences.getBoolean(KEY_GRANTED, false)

    fun setGranted(granted: Boolean) {
        preferences.edit().putBoolean(KEY_GRANTED, granted).apply()
    }
}

File: ios/Services/AnalyticsConsentStore.swift
import Foundation

final class AnalyticsConsentStore {
    private let defaults: UserDefaults
    private let key = "analytics.consent.granted"

    init(defaults: UserDefaults = .standard) {
        self.defaults = defaults
    }

    var isGranted: Bool {
        defaults.bool(forKey: key)
    }

    func setGranted(_ granted: Bool) {
        defaults.set(granted, forKey: key)
    }
}
